### ISSUE TITLE
Sometimes only a part of the map is generated

### ISSUE THREAD

**Issue by SaulGoodman1337** [Hypfer/Valetudo#41]
Sometimes it happens that not the whole map is generated, but only a part of it.

I have attached a video to it. 
https://youtu.be/yGNcIR5bgXI

The problem occurs not only in Homeassistant (set up as a camera), but also when I access the maps directly via URL. After a few refreshes only half of the image comes up every now and then.

**Comment by danielpondruff** [comment]
any update here ?

**Comment by kgc210** [comment]
I have the same issue but I don't think this is a bug in Valetudo it's self, more how Home Assistant renders the image.
The reason I say this is that if you download every png that gets generated in the tmp folder they are all complete images.

**Comment by leonardpitzu** [comment]
i have the same issue - it's Valetudo as i use it to control the vacuum directly (no home assistant integration, etc.)

**Comment by danielpondruff** [comment]
> I have the same issue but I don't think this is a bug in Valetudo it's self, more how Home Assistant renders the image.
> The reason I say this is that if you download every png that gets generated in the tmp folder they are all complete images.

Nope. this is not Home assistant releated. I have this issue also when i get the map with my webbrowser directly from bei vacuum. 

> i have the same issue - it's Valetudo as i use it to control the vacuum directly (no home assistant integration, etc.)

i thinks this is a valetudo issue, too

**Comment by leonardpitzu** [comment]
there seems to be some size limiting thing - during cleanup the map is correctly generated. after a certain area is covered a new map is generated. i guess it has to do with the new map format discussed in #44 - the map has to be pieced together so we can see the entire map...

**Comment by ideasman69** [comment]
I've noticed the same as @JohnRev. Like in this picture, the areas highlighted in red have been completed and did have data earlier, but now has disappeared...

![2019-01-19_11-13-23](https://user-images.githubusercontent.com/19260137/51419116-1294af80-1bdc-11e9-91e2-29de5295b221.png)

I just rooted my gen1 yesterday accoring to [this guide](https://hackmd.io/s/HkwF33oWE). Firmware v3514, no errors from firmwarebuilder and patching of rrlog says it was successful.
